spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -263,4 +281,12 @@ public SubscriberStub createSubscriberStub() {
 		}
 	}
 
+	@Override
+	public PubSubAcknowledger createAcknowledger() {
+		DefaultPubSubAcknowledger acknowledger =
+				new DefaultPubSubAcknowledger(createSubscriberStub());
+		acknowledger.setAsync(this.asyncAck);
```

Comment:
Since the only place this method is currently used is in the `PubSubSubscriberTemplate` constructor, making `asyncAck` a settable property could be pretty confusing because if you change it after the application context is loaded, it won't actually affect anything.

Perhaps adding a `createAsyncAcknowledger` is better, and the caller should really decide which one they want rather than depending on the configuration in the `DefaultSubscriberFactory`. It almost feels that this whole method is unnecessary because the template can simply `DefaultPubSubAcknowledger` on its own by first calling `createSubscriberStub()`.

Reply:
IMO, the `createAsyncAcknowledger()` is not what could be a part of the `SubscriberFactory` contract.
That's really an implementation details.

And yes, you are right that `template` can mock our `acknowledger` API using direct access. That's why I didn't see the reason of this `createAcknowledger()` contract in the beginning at all...

Reply:
Let's just remove `createAcknowledger` and do it in 1.0.x as well.